src/System.Net.Http/tests/UnitTests/DigestAuthenticationTests.cs

```diff
@@ -40,5 +40,27 @@ public static IEnumerable<object[]> DigestResponse_Challenge_TestData()
             yield return new object[] { "=value1,key2=,", s_emptyStringList, s_emptyStringList };
             yield return new object[] { "key1\tm= value1", s_emptyStringList, s_emptyStringList };
         }
+
+        [Theory]
+        [InlineData("realm=\"NetCore\", nonce=\"qMRqWgAAAAAQMjIABgAAAFwEiEwAAAAA\", qop=\"auth\", stale=false")]
+        public async void DigestResponse_AuthToken_Succeeds(string response)
+        {
+            NetworkCredential credential = new NetworkCredential("foo","bar");
+            AuthenticationHelper.DigestResponse digestResponse = new AuthenticationHelper.DigestResponse(response);
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, "http://microsoft.com/");
+
+            Assert.True(await AuthenticationHelper.TrySetDigestAuthToken(request, credential, digestResponse, HttpKnownHeaderNames.ProxyAuthorization).ConfigureAwait(false));
+        }
+
+        [Theory]
+        [InlineData("nonce=\"qMRqWgAAAAAQMjIABgAAAFwEiEwAAAAA\", qop=\"auth\", stale=false")]
```

Comment:
You can combine this with the previous test method, the point of Theory is to have multiple Inlinedata for the same test code.. This is waster duplication.

Reply:
The following suite of tests make this complete:

1. Realm missing, returns false
2. Nonce missing, returns false
3. domain, opaque, stale, algorithm, qop options, auth-param missing returns true

Reply:
You can have just this single test instead of the two different test methods..

```c#
[Theory]
[MemberData(nameof(DigestResponse_AuthToken_TestData))]
public async void DigestResponse_AuthToken_Fails(string challenge, bool expected)
{
     NetworkCredential credential = new NetworkCredential("foo","bar");
     AuthenticationHelper.DigestResponse digestResponse = new AuthenticationHelper.DigestResponse(challenge);
     HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, "http://microsoft.com/");
 
     Assert.Equal(expected, await AuthenticationHelper.TrySetDigestAuthToken(request, credential, digestResponse, HttpKnownHeaderNames.ProxyAuthorization).ConfigureAwait(false));
 }

public static IEnumerable<object[]> DigestResponse_AuthToken_TestData()
{
    // realm missing
    yield return new object[] { "nonce=\"qMRqWgAAAAAQMjIABgAAAFwEiEwAAAAA\"", false };
    // nonce missing
    yield return new object[] { "realm=\"testrealm\"", false };
    // domain, opaque, stale, algorithm, qop-options, auth-param missing
    yield return new object[] { "realm=\"testrealm\"", nonce=\"qMRqWgAAAAAQMjIABgAAAFwEiEwAAAAA\", true };
}
```